packages/cli/src/transformers.test.ts

```diff
@@ -24,1266 +26,729 @@ import {
+        "
+      `);
     });
 
-    expect(
-      await compile({
-        format: 'module',
-        environment,
-        transformer: getRequireTransformer({
-          typeChecker: environment.typeChecker,
-          system: environment.system,
-          baseDirectory: '/',
-        }),
-      }),
-    ).toMatchInlineSnapshot(`
-      ""use strict";
-      const foo = 'bar';
-      "
-    `);
+    it('does not add a shim when `require` refers to a variable in scope', async () => {
+      expect(files['require-in-scope.js']).toMatchInlineSnapshot(`
+        "// @ts-expect-error - \`require\` is an existing global.
+        const require = (_module) => undefined;
+        require('module');
+        export {};
+        "
+      `);
+    });
   });
 });
 
 describe('getImportMetaTransformer', () => {
-  it('adds a shim when using `import.meta.url`', async () => {
-    const environment = getVirtualEnvironment({
-      files: {
-        '/index.ts': `
-          // @ts-expect-error - 'import.meta' is not allowed.
-          console.log(import.meta.url);
-        `,
-      },
+  describe('when targeting `commonjs`', () => {
+    let files: Record<string, string>;
+
+    beforeAll(() => {
+      const compiler = createCompiler(getFixture('globals'));
+      files = compiler(
+        'commonjs',
+        getImportMetaTransformer({
+          typeChecker: compiler.typeChecker,
+          system: sys,
+        }),
+      );
+    });
+
+    it('adds a shim when using `import.meta.url`', async () => {
+      expect(files['import-meta.js']).toMatchInlineSnapshot(`
+        ""use strict";
+        var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
+            if (k2 === undefined) k2 = k;
+            var desc = Object.getOwnPropertyDescriptor(m, k);
+            if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
+              desc = { enumerable: true, get: function() { return m[k]; } };
+            }
+            Object.defineProperty(o, k2, desc);
+        }) : (function(o, m, k, k2) {
+            if (k2 === undefined) k2 = k;
+            o[k2] = m[k];
+        }));
+        var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
+            Object.defineProperty(o, "default", { enumerable: true, value: v });
+        }) : function(o, v) {
+            o["default"] = v;
+        });
+        var __importStar = (this && this.__importStar) || function (mod) {
+            if (mod && mod.__esModule) return mod;
+            var result = {};
+            if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
+            __setModuleDefault(result, mod);
+            return result;
+        };
+        Object.defineProperty(exports, "__esModule", { value: true });
+        const $shims = __importStar(require("@ts-bridge/shims"));
```

Comment:
Potentially unrelated to this PR, but do we need to use the `importStar` pattern for the shims? Seems like it adds a lot of code.

Reply:
The shims are added as `import * as $shims from '@ts-bridge/shims'`, and TypeScript changes that to use the `__importStar` helper. We can refactor it to use named imports, but like you said it's not related to this PR.